kustomize.go

```diff
@@ -143,15 +143,15 @@ func (r *Runner) KustomizeBuild(srcDir string, tempDir string, opts ...Kustomize
 		r.Logf("Failed to parse `kustomize version` output: %v\nFalling-back to the kustomize v4 mode...", err)
 	}
 
-	if major > 3 {
-		kustomizeArgs = append(kustomizeArgs, "--load-restrictor=LoadRestrictionsNone")
+	if major <= 3 {
```

Comment:
```suggestion
	if major <= 3 && err == nil {
```

@elliottminns Hey! Thanks a lot for submitting this pull request ❤️ 
I believe you need to exclude the case where it failed to parse any version here.
Otherwise, a parse failure result in going this code path and the end result is you're still falling back to the pre-v4 flags, which is the opposite of what you wanted to do according to your PR description.